lib/secretFactory.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const BaseFactory = require('./baseFactory');
+const Secret = require('./secret');
+const iron = require('iron');
+const nodeify = require('./nodeify');
+const hoek = require('hoek');
+// Get symbols for private fields
+const password = Symbol();
+
+let instance;
+
+/**
+ * Seal secret
+ * @method sealSecret
+ * @param  {String}   secret      Secret to seal
+ * @param  {String}   pw          Password to seal with
+ * @return {Promise}
+ */
+const sealSecret = (secret, pw) =>
+    nodeify.withContext(iron, 'seal', [secret, pw, iron.defaults]);
+
+/**
+ * Unseal secret
+ * @method  unsealSecret
+ * @param  {String}   secret      Secret to unseal
+ * @param  {String}   pw          Password to unseal with
+ * @return {Promise}
+ */
+const unsealSecret = (secret, pw) =>
+    nodeify.withContext(iron, 'unseal', [secret, pw, iron.defaults]);
+
+class SecretFactory extends BaseFactory {
+    /**
+     * Construct a SecretFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore     Object that will perform operations on the datastore
+     * @param  {String}    config.password      Password for encryption operations
+     */
+    constructor(config) {
+        super('secret', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Instantiate a Secret class
+     * @method createClass
+     * @param  config
+     * @return {Secret}
+     */
+    createClass(config) {
+        const c = hoek.applyToDefaults(config, { password: this[password] });
+
+        return new Secret(c);
+    }
+
+    /**
+     * Create a secret model
+     * Need to seal the secret before saving
+     * @method create
+     * @param  {Object}     config
+     * @param  {String}     config.pipelineId      Pipeline Id
+     * @param  {String}     config.name            Secret name
+     * @param  {String}     config.value           Secret value
+     * @param  {Boolean}    config.allowInPR       Whether this secret can be shown in PR builds
+     * @return {Promise}
+     */
+    create(config) {
+        return sealSecret(config.value, this[password])
+            .then(secret => super.create(hoek.applyToDefaults(config, { value: secret })));
+    }
+
+    /**
+     * Get a record based on id
+     * @method get
+     * @param  {Mixed}   config    The configuration from which an id is generated or the actual id
+     * @return {Promise}
+     */
+    get(config) {
```

Comment:
This does little more than redefine all the code in the base class. Why not utilize the base class code, and do the data wrangling with the result?

```
return super.get(config).then(secret => unseal(secret.value)).then(unsealed => {secret.value = unsealed; return secret; } );
```